snapcraft/tests/test_plugin_tar_content.py

```diff
@@ -0,0 +1,83 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os.path
+import tarfile
+from snapcraft.plugins.tar_content import TarContentPlugin
+from snapcraft.tests import TestCase
+
+
+class TestTarContentPlugin(TestCase):
+
+    def setUp(self):
+        super().setUp()
+        # setup the expected target dir in our tempdir
+        self.build_prefix = 'parts/tarContent/build/'
+        os.makedirs(self.build_prefix)
+        self.install_prefix = 'parts/tarContent/install/'
+        os.makedirs(self.install_prefix)
+
+    def test_dest_abs_path_raises_exception(self):
+        class Options:
+            source = '.'
+            destination = '/destdir1'
+        # ensure that a absolute path for a destination directory
+        # raises an exception
+        with self.assertRaises(ValueError) as raised:
+            TarContentPlugin('tarContent', Options())
+
+        self.assertEqual(raised.exception.__str__(),
+                         "path \"/destdir1\" must be relative")
+
+    def test_build_destination_dir_exists(self):
+        class Options:
+            source = '.'
+            destination = 'destdir1'
+        TarContentPlugin('tarContent', Options())
+
+        self.assertTrue(
+            os.path.exists(os.path.join(self.build_prefix, 'destdir1')))
+
+    def test_install_destination_dir_exists(self):
+        class Options:
+            source = os.path.join('src', 'test.tar')
+            destination = 'destdir1'
+
+        # create tar file for testing
+        os.mkdir('src')
+        fileToTar = os.path.join('src', 'test.txt')
+        open(fileToTar, 'w').close()
+        tar = tarfile.open(os.path.join('src', 'test.tar'), "w")
+        tar.add(fileToTar)
+        tar.close()
+
+        t = TarContentPlugin('tarContent', Options())
+        t.build()
```

Comment:
one letter variables are not common in practice. I would call this plugin instead of t, or chain the build call, so you don't have to assign it to anything:
TarContentPlugin('tarContent', Options()).build()